### DISCUSSION TITLE
question: How to restore from a backup?

### DISCUSSION THREAD

**Discussion by abmusse** [AsamK/signal-cli#1297] [Q&A] [answered]
I'm looking to backup my current `signal-cli` data and restore onto a new machine.

Is the following process sufficient to backup and restore?

1) Backup `~/.local/share/signal-cli` 
2) Install `signal-cli` onto the new machine
3) Restore `~/.local/share/signal-cli` onto the new machine

I don't want to have to go through `signal-cli` initialization again on the new machine.

**Comment by AsamK** [comment, marked as answer]
Yes all signal-cli data is stored in that folder.

**Reply by abmusse** [reply]
Awesome!

Will give that a shot

**Reply by interdisziplina** [reply]
@abmusse Did this work properly and did you have to configure anything else?

**Reply by abmusse** [reply]
Yup it did end up working for me, keep in mind I tested this while ago though so YMMV.
